db/silkworm/db/access_layer.hpp

```diff
@@ -24,11 +24,38 @@
 #include <vector>
 
 #include <silkworm/chain/config.hpp>
+#include <silkworm/common/magic_enum.hpp>
 #include <silkworm/db/chaindb.hpp>
 #include <silkworm/db/util.hpp>
 #include <silkworm/types/account.hpp>
 #include <silkworm/types/block.hpp>
 
+namespace silkworm::rlp {
+
+class decode_error : public std::exception {
```

Comment:
I'd name it `DecodingError` for consistency with [Google's style guide](https://google.github.io/styleguide/cppguide.html#Type_Names)